### ISSUE TITLE
Add horizontal scrollbar to System Information graphs

### ISSUE THREAD

**Issue by SanderBouwhuis** [processhacker/processhacker#659]
Currently, how far back in history you can see changes depends on how large you make the window. So, someone with a large screen can see more than someone with a small screen. That is ridiculous!

A small window (look back a tiny bit):
![ProcessHacker_YWxHYDBgYP](https://user-images.githubusercontent.com/10495173/90378560-ce98b380-e079-11ea-9b34-6093fd430928.png)

A large window (look back quite far):
![ProcessHacker_9ZFWhacXXJ](https://user-images.githubusercontent.com/10495173/90378608-dd7f6600-e079-11ea-835e-181bd275453a.png)

Please add a horizontal scrollbar so that we can always look back.

**Comment by dmex** [comment]
>  someone with a large screen can see more than someone with a small screen. That is ridiculous!

Yep, welcome to win32k limitations and it's the same issue with graphs in Task Manager and Process Explorer.

The window resize messages combined with mouse input messages and the scrollbar control and subclass messages causes delayed input and window flicker/repainting issues because the amount of controls and messages either exceed the maximum limit of the message queue or take too long during processing.

You won't generally notice too many input and redrawing issues on a good machine but Task Manager applications like Process Hacker are required to execute on machines under heavy utilization and/or already having some major performance issues (The only time most people even use a Task Manager type application is when their machine is having issues so it needs to work under extreme conditions).

In cases of extreme system load the window and graphs can sometimes take a minute to render and the extra controls would make it unusable - Process Explorer has the exact same problem and its so much worse with Task Manager because of its dui70 rendering.

> How far back in history you can see changes depends on how large you make the window

For example; A screen with 1200x760 resolution shows 20 minutes of history, times that resolution by the number of visible graphs and you have 28,800 data samples visible on the screen.

The memory and other utilisation generated by the redrawing and sampling makes every application unusable for long durations even in the best conditions. You need a good machine and good graphics card for running a larger display so tying the graphs to the display resolution always ensures you won't have issues with rendering and can handle the costs of the larger graphs, larger amounts of samples and larger memory/utilisation cost.

I don't see a better way of handling these cases without also abandoning support for execution under extreme conditions which defeats the entire purpose of a Task Manager style application. 

So when I've needed the extra sampling; I've logged the samples into a file and use a separate executable to render the data since this isolates the samples/rendering/performance cost... I could commit this stuff if its something that might be useful?

**Comment by SanderBouwhuis** [comment]
Logging to a file? YES PLEASE!!!
This is actually what I wanted in the first place!

I have some processes I run over long periods of time, so if I could get PH to log to a file (buffered writing once per minute or so) that would be great. Especially if this would work for every graph (network, disk, CPU, RAM).

I guess if this would be a plugin, then no one is going to feel the cost/overhead unless you turn it on.

**Comment by dmex** [comment]
How well does the typedef command work for logging data to a file?

For example open command prompt and execute:
 `typeperf "\Process(chrome#2)\% Processor Time"`
`typeperf "\Processor(_Total)\% Processor Time"`

Listing counters:
`typeperf -q`

Search and save counters to file:
`typeperf -qx STRING -o counters.txt`

List all counters from previous file and log into csv file (supports csv, tsv, bin and SQL)
`typeperf.exe -f CSV -o test.csv -cf  counters.txt`

**Comment by dmex** [comment]
I deleted my first comment by accident when updating it with new info :(

> lease add a horizontal scrollbar so that we can always look back.

I'm testing horizontal scrollbars for the graphs over here:
https://github.com/processhacker/plugins-extra/tree/master/GraphExplorerPlugin

> Logging to a file?

This might only work by logging into a sqlite database with options to export the data into various formats?

**Comment by SanderBouwhuis** [comment]
1. So, the plugin adds a horizontal scrollbar to the graphs?
2. Do you have a release package we can download to test/try this?
3. Maybe it would be possible to export the raw data by pressing the right mouse button and getting a popup menu where you can choose 'Export raw data'?

**Comment by dmex** [comment]
1) It's a new type of graph using the treelist control. The scrollbars are included with the treelist.
2) The plugin is for testing stuff and not production ready.
3) Export is possible but a separate issue?

**Comment by SanderBouwhuis** [comment]
Ok, I'll eagerly await the horizontal scrollbar plugin! 👍
